flasc/energy_ratio/energy_ratio.py

```diff
@@ -655,11 +666,33 @@ def _get_energy_ratio_single_wd_bin_bootstrapping(
     if N <= 1:
         results_array = np.array([energy_ratio_nominal] * 3, dtype=float)
     else:
+
+        # If using block-bootstrapping, set up blocks
+        if block_bootstrapping:
+            block_list = list(range(num_blocks))  # List of all block names
```

Comment:
I haven't tried this, but I wonder how things work if you have 10 samples and split it over 3 blocks. It should split it into 3 vectors with length 4, 3 and 3, respectively. And if you have 8 samples and split it into 3 blocks, it should have three vectors with length 3, 3 and 2. Do those work out, currently? My thought with the current line of code ` block_length = int(len(block_indices) / num_blocks)` was that this may not be resolved properly.

Maybe consider `np.split_array` is a good function for this. For example
```
sample_indices = np.arange(0, num_samples, dtype=int)
block_indices = np.array_split(sample_indices, num_blocks)
```

Reply:
I like that, I'll give it a shot!

Reply:
hi @Bartdoekemeijer  I remember what I did here, I start out by assigning everything to the first (0th block) with this line:

```
block_labels = np.zeros(len(block_indices)).astype(int) # Labels to assign each index
```

then I assign the blocks here:

```
for b_i in range(num_blocks):
    block_labels[block_length * b_i:block_length * (b_i+1)] = b_i
```

If the things don't divide evenly the leftovers get assigned to the first (0th block) which I guess could make that block a little bigger than the others but it gives some wrapping around, which I thought could be desirable.  But maybe your method is still preferable since it makes sure things are more even